controllers/operator_controller.go

```diff
@@ -145,11 +154,47 @@ func (r *OperatorReconciler) reconcile(ctx context.Context, op *operatorsv1alpha
 	return ctrl.Result{}, nil
 }
 
+func (r *OperatorReconciler) generateExpectedBundleDeployment(o operatorsv1alpha1.Operator, bundlePath string) (*rukpakv1alpha1.BundleDeployment, error) {
+	dep := &rukpakv1alpha1.BundleDeployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: o.GetName(),
+		},
+		Spec: rukpakv1alpha1.BundleDeploymentSpec{
+			//TODO: Don't assume plain provisioner
+			ProvisionerClassName: "core-rukpak-io-plain",
+			Template: &rukpakv1alpha1.BundleTemplate{
+				ObjectMeta: metav1.ObjectMeta{
+					// TODO: Remove
+					Labels: map[string]string{
+						"app": "my-bundle",
+					},
+				},
+				Spec: rukpakv1alpha1.BundleSpec{
+					Source: rukpakv1alpha1.BundleSource{
+						// TODO: Don't assume image type
+						Type: rukpakv1alpha1.SourceTypeImage,
+						Image: &rukpakv1alpha1.ImageSource{
+							Ref: bundlePath,
+						},
+					},
+
+					//TODO: Don't assume registry provisioner
+					ProvisionerClassName: "core-rukpak-io-registry",
+				},
+			},
+		},
+	}
+
+	if err := ctrl.SetControllerReference(&o, dep, r.Scheme); err != nil {
+		return nil, err
+	}
+	return dep, nil
+}
+
 // SetupWithManager sets up the controller with the Manager.
 func (r *OperatorReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	r.resolver = resolution.NewOperatorResolver(mgr.GetClient(), resolution.HardcodedEntitySource)
-
 	err := ctrl.NewControllerManagedBy(mgr).
+		Owns(&rukpakv1alpha1.BundleDeployment{}).
```

Comment:
Convention/nit/my opinion: Always declare `For` as the first watch. Its the primary CR type. Then Owns, then Watches.

Sort of ordering from most coupling (self) to least coupling (things I watch, but don't own).

Reply:
Yeah, that makes sense to me too. I'll update it.